### DISCUSSION TITLE
Automatic Login

### DISCUSSION THREAD

**Discussion by xeddog** [OctoFarm/OctoFarm#581] [Feature Request]
I would like to see an option to allow automatic login.   I have a single user machine instance,  and every time I launch OctoFarm I am presented with the login screen.  I have used the "Remember Me" button, so the userid and password fields are already filled in.  So how about an option to go ahead and "press enter" so to speak, and complete the login?

**Comment by davidzwa** [comment]
Hi @xeddog this option already exists. Please visit the system page for disabling login - it sounds different but it does what you want.

This is the subsection.
![image](https://user-images.githubusercontent.com/6005355/118407951-7849b700-b683-11eb-8f0e-50fd9a10a0a2.png)

Of course, let us know if this was not what you had in mind, or if you have questions still

**Comment by NotExpectedYet** [comment]
> I would like to see an option to allow automatic login. I have a single user machine instance, and every time I launch OctoFarm I am presented with the login screen. I have used the "Remember Me" button, so the userid and password fields are already filled in. So how about an option to go ahead and "press enter" so to speak, and complete the login?

Does the remember me button not auto log you in? It should be keeping your cookie valid and not redirecting you to the login page. There's also as David said but that's if you don't want a login at all. The remember me button could probably do with a customised timeout tbh, I believe it's only set at 7 days before you have to re-login.

**Reply by davidzwa** [reply]
Reading OP's message again, I think he's confusing autofill with remember me. And yeah that suggests Remember Me is not working.

**Reply by NotExpectedYet** [reply]
That's what I got from his message anyway.

**Comment by xeddog** [comment]
I was not looking for eliminating a login all together, although for me that would also be ok since I am the only person in the household that will ever use OctoFarm.   But what is happening now is that after OctoFarm is up and running, and then launch the browser (Firefox for me), I am presented with a panel that says to register or login.  When I click Login, I get another panel, a login panel, that already has the userid and password fields filled in from a prior "remember me".  That requires one last click to get logged in so I can use the application.   

What I was actually looking for is an option so that after the initial registration/login,  the "remembered" userid and password would be used to automagically complete the login so I am presented with the dashboard screen.  

As I mentioned, personally I am ok with just eliminating the login all together, but maybe other users might find auto-login useful.  Are there any commands, statistics, or other information collected where a userid would be required????

**Reply by davidzwa** [reply]
I have to repeat: checking "Remember Me" is not the same as getting username and password autofilled by the browser.
If "Remember Me" works it should skip the login form altogether, again that has little to do with filling input fields like username/password. 

The only case Remember Me is not going to work is when the OctoFarm server is restarted, but I dont guess you do that. Please confirm this. If you're not restarting OctoFarm before logging in, there's some bug or something James and me have to look at and fix.

**Reply by xeddog** [reply]
I typically shutdown all of my computers (except RPi's) every night just to save a few electrons, printers too if they are not busy, and then bring them back up the next morning when/if I want to use them again.  So in that context, I restart OctoFarm every morning, as opposed to System>Server>Restart (which I have never used).

**Reply by NotExpectedYet** [reply]
The Remember Me code doesn't persist in the database currently so that explains why your having to re-login everytime. It expects an always on server. We could solve this with persisting to the database in a later release but it sounds like for now disabling login will at least work for your use case.

**Comment by xeddog** [comment]
Yeah, disabling login will work fine for me.  But that brings me back to the question I asked earlier which is, essentially, what is the purpose of the login?  Are there any additional features or commands that require a userid?  Debugging or logging???  Can a user be "controlled" by limiting access to a defined group of printers???  Future ??????

**Reply by NotExpectedYet** [reply]
It's a basic implementation to lock off the interface for people who want to for now. Currently the first user is admin and any subsequent user becomes user grouped. Admins are the only ones allowed to edit server settings and add remove printers currently. 

In future I want to allow more granular permissions to this stack like uploading files, starting prints, managing spools, user creation and more similar to what you mentioned. 

Currently when you disable the login it will treat everything you do as an admin user with all I mentioned above so there's no difference in a single user with login and login been turned off. Hope that helps, sorry as well must have missed your question in my haste.
